benchmark/fs/bench-readdir.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+const fs = require('fs');
+const print = process._rawDebug;
+
+const ITER = 1e3;
+var t = process.hrtime();
+
+(function r(n) {
+  if (--n <= 0)
+    return completed();
+  fs.readdir(__dirname + '/../../lib/', function() {
+    r(n);
+  });
+}(ITER));
+
+function completed() {
+  t = process.hrtime(t);
+  print(((t[0] * 1e6 + t[1] / 1e3) / ITER).toFixed(2) + ' us/op');
+}
```

Comment:
Here's the flame graph for the benchmark using the patch: https://cldup.com/aYznQq9OKJ.svg
and flame graph without the patch: https://cldup.com/XUtalQpRpm.svg

There's almost no idle time in this test. GC is also not an issue. For with and w/o the patch GC reports practically nothing is going on. The directory lookup my have been optimized out by the system, and the test is going to be bias towards machines w/ SSDs anyway.